src/client/src/consumer.rs

```diff
@@ -564,38 +571,159 @@ static MAX_FETCH_BYTES: Lazy<i32> = Lazy::new(|| {
     max_bytes
 });
 
+#[derive(Debug, Clone)]
+pub struct SmartStreamModule {
+    binary: Vec<u8>,
+}
+
+impl SmartStreamModule {
+    /// Instantiate a SmartStream from raw WASM binary
+    pub fn from_binary<T: Into<Vec<u8>>>(binary: T) -> Self {
+        Self {
+            binary: binary.into(),
+        }
+    }
+
+    /// Instantiate a SmartStream from a base64-encoded string
+    pub fn from_base64<S: Into<String>>(string: S) -> Result<Self, FluvioError> {
```

Comment:
This seems too premature to add as supporting API.  This conversion can be done by caller.